atc/db/notifications_bus.go

```diff
@@ -52,14 +53,26 @@ type notificationsBus struct {
 	wg *sync.WaitGroup
 }
 
+var notificationBusQueueSize = 10000
+
+func SetNotificationBusQueueSize(size int) error {
+	if size <= 0 {
+		return nil
+	}
+	if size < 1000 || size > 1000000 {
```

Comment:
what is the reasoning behind these two numbers? What will happen if the size is too small or too big?

I would like to know the gains and cost while setting the queue size with different number. If setting default to 10000 works for most cases, we should avoid adding this flag.

Reply:
The reason why I added this parameter is I don't know what value is proper, so that after we apply this patch to our prod, the queue size must be tunable. If queue size is too small, running builds will not be able to refreshed in time (note that, only impact running builds); if queue size is too big, it will consume a lot of ATC memories. So I set a wide range which allows us to tune the parameter with enough flexibility.